### PR TITLE
[Vulnerability Disclosure and security.txt] guidance about redirects and bump date

### PR DESCRIPTION
**Pull request by OllieJC**
- After a few implementations with different services teams, this change adds a little guidance about the order in which redirect mechanisms should be implemented.
- Bumps review date.

### CODE REVIEW

File: source/standards/vulnerability-disclosure.html.md.erb [hunk 1]
```diff
@@ -31,7 +31,12 @@ You should use <https://vdp.cabinetoffice.gov.uk/.well-known/security.txt>
 in either:
 
 - the origin for your site's `/.well-known/security.txt`
-- the destination of a 302 redirect for `/.well-known/security.txt`
+- the destination of a redirect for `/.well-known/security.txt`
+  - note - try implementing the redirect mechanism in the following order:
+    1. Server-side redirect (302 status and `Location` header in response)
+    2. Client-side HTML (meta `http-equiv=refresh` tag in the head)
+    3. Client-side JavaScript redirect (`window.location.href`) - this won't
+work if JavaScript is disabled, so you should display a link as well
```
Comment (jamietanna) [inline source/standards/vulnerability-disclosure.html.md.erb:39]: I'd say removing the linebreak may be best:
```suggestion
    3. Client-side JavaScript redirect (`window.location.href`) - this won't work if JavaScript is disabled, so you should display a link as well
```

File: source/standards/vulnerability-disclosure.html.md.erb [hunk 2]
```diff
@@ -31,7 +31,12 @@ You should use <https://vdp.cabinetoffice.gov.uk/.well-known/security.txt>
 in either:
 
 - the origin for your site's `/.well-known/security.txt`
-- the destination of a 302 redirect for `/.well-known/security.txt`
+- the destination of a redirect for `/.well-known/security.txt`
+  - note - try implementing the redirect mechanism in the following order:
+    1. Server-side redirect (302 status and `Location` header in response)
```
Comment (jamietanna) [inline source/standards/vulnerability-disclosure.html.md.erb:36]: have we checked what happens when this renders? I know some Markdown processors may have problems with newlines being not present between things, but could be it's safe as it's inside a list

Comment (OllieJC) [reply source/standards/vulnerability-disclosure.html.md.erb]: I had not 🤦‍♂️ Made changes, here's how it looks:

image

**Comment by OllieJC** [comment]
Hey @jamietanna - could I get your thoughts/approval on the changes, please?

**Comment by OllieJC** [comment]
Hey @jamietanna - could I get your thoughts/approval on the changes, please?

File: source/standards/vulnerability-disclosure.html.md.erb [hunk 3]
```diff
@@ -31,7 +31,15 @@ You should use <https://vdp.cabinetoffice.gov.uk/.well-known/security.txt>
 in either:
 
 - the origin for your site's `/.well-known/security.txt`
-- the destination of a 302 redirect for `/.well-known/security.txt`
+- the destination of a redirect for `/.well-known/security.txt`
+
+A note on the redirect mechanism, try implementing in the following order to
+ensure the best capability with all user agents:
```
Comment (jamietanna) [inline source/standards/vulnerability-disclosure.html.md.erb:37]: I personally wouldn't line-break these, but not sure what the GDS Way is for this :upside_down_face: 

```suggestion
A note on the redirect mechanism, try implementing in the following order to ensure the best capability with all user agents:
```

File: source/standards/vulnerability-disclosure.html.md.erb [hunk 4]
```diff
@@ -31,7 +31,15 @@ You should use <https://vdp.cabinetoffice.gov.uk/.well-known/security.txt>
...
+
+1. Server-side redirect (302 status and `Location` header in response)
+2. Client-side HTML (meta `http-equiv=refresh` tag in the head)
+3. Client-side JavaScript redirect (`window.location.href`) - this won't work
+if JavaScript is disabled, so you should display a link as well
```
Comment (jamietanna) [inline source/standards/vulnerability-disclosure.html.md.erb:42]: I personally wouldn't line-break these, but not sure what the GDS Way is for this :upside_down_face: 

```suggestion
3. Client-side JavaScript redirect (`window.location.href`) - this won't work if JavaScript is disabled, so you should display a link as well
```

File: source/standards/vulnerability-disclosure.html.md.erb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Vulnerability Disclosure and security.txt
-last_reviewed_on: 2021-04-22
+last_reviewed_on: 2021-11-02
```
Comment (jamietanna) [inline source/standards/vulnerability-disclosure.html.md.erb:3]: Could probably be updated now :smile:

**Review by jamietanna — COMMENTED** [review]
Couple of tweaks but Looks Good To Me :rocket: 

I'm not a committer, so we'll need someone else to approve

**Review by timblair — APPROVED** [review]
LGTM. I've bumped the review date to today.

File: source/standards/vulnerability-disclosure.html.md.erb (hunk 4 above)
Comment (OllieJC) [reply source/standards/vulnerability-disclosure.html.md.erb:42]: Thanks! I personally like the line breaks for reviewing in terminal but I defo take on board that others will dislike or find the line breaks confusing. There's [some mention](https://gds-way.cloudapps.digital/manuals/programming-languages/python/python.html#maximum-line-length-of-120-characters) on max line length for _code_, but not markdown. I'll see where's best to flag this, perhaps I'll raise an issue for comments. I'm going to resolve these and merge from Tim's approval but will follow up on md line breaking :)

**Pull request merged**